tests/storage/blocksd_test.py

```diff
@@ -1244,6 +1251,117 @@ def test_create_volume_with_new_bitmap(
     ]
 
 
+@requires_root
+@pytest.mark.root
+@pytest.mark.xfail(reason='Bitmaps in base volume not considered for merge '
+                          'if added after top volume is created.')
+def test_merge_with_bitmap(
+        tmp_storage, tmp_repo, fake_access, fake_rescan, tmp_db, fake_task,
+        fake_sanlock, monkeypatch):
+    sd_id = str(uuid.uuid4())
+
+    dev = tmp_storage.create_device(10 * GiB)
+    lvm.createVG(sd_id, [dev], blockSD.STORAGE_UNREADY_DOMAIN_TAG, 128)
+    vg = lvm.getVG(sd_id)
+
+    dom = blockSD.BlockStorageDomain.create(
+        sdUUID=sd_id,
+        domainName="domain",
+        domClass=sd.DATA_DOMAIN,
+        vgUUID=vg.uuid,
+        version=4,
+        storageType=sd.ISCSI_DOMAIN)
+
+    sdCache.knownSDs[sd_id] = blockSD.findDomain
+    sdCache.manuallyAddDomain(dom)
+
+    dom.refresh()
+    dom.attach(tmp_repo.pool_id)
+
+    img_id = str(uuid.uuid4())
+    base_vol_uuid = str(uuid.uuid4())
+    top_vol_uuid = str(uuid.uuid4())
+
+    # Create base volume.
+    dom.createVolume(
+        imgUUID=img_id,
+        capacity=1 * GiB,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.SPARSE_VOL,
+        diskType='DATA',
+        volUUID=base_vol_uuid,
+        desc="Test volume",
+        srcImgUUID=sc.BLANK_UUID,
+        srcVolUUID=sc.BLANK_UUID,
+        bitmap="new-bitmap")
+
+    base_vol = dom.produceVolume(img_id, base_vol_uuid)
+
+    # Create top volume with bitmaps.
+    dom.createVolume(
+        imgUUID=img_id,
+        capacity=1 * GiB,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.SPARSE_VOL,
+        diskType='DATA',
+        volUUID=top_vol_uuid,
+        desc="Test top volume",
+        srcImgUUID=base_vol.imgUUID,
+        srcVolUUID=base_vol.volUUID,
+        add_bitmaps=True)
+
+    top_vol = dom.produceVolume(img_id, top_vol_uuid)
+
+    # Create leaf volume.
+    dom.createVolume(
+        imgUUID=img_id,
+        capacity=1 * GiB,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.SPARSE_VOL,
+        diskType='DATA',
+        volUUID=str(uuid.uuid4()),
+        desc="Test top volume",
+        srcImgUUID=top_vol.imgUUID,
+        srcVolUUID=top_vol.volUUID,)
+
+    host_id = 0
+    subchain_info = dict(sd_id=base_vol.sdUUID,
+                         img_id=base_vol.imgUUID,
+                         base_id=base_vol.volUUID,
+                         top_id=top_vol.volUUID,
+                         base_generation=0)
+    subchain = merge.SubchainInfo(subchain_info, host_id)
+
+    # Nearly fill the volume with data.
+    top_vol.prepare()
+    data_size = GiB - 5 * MiB
```

Comment:
The initial size of the base volume is 2.5g. We write here 1g, and we write 100 2k
bitmpas that takes 200K plus some space for the bitmap directory. How this can fail
with ENOSPC?